### ISSUE TITLE
Add an interface or a library to enables providers implement downward API

### ISSUE THREAD

**Issue by helayoty** [virtual-kubelet/virtual-kubelet#1032]
_Originally posted by @cpuguy83 in https://github.com/virtual-kubelet/virtual-kubelet/issues/999#issuecomment-1233336562_

**Comment by pires** [comment]
Also see #123 and #998

**Comment by tskillian** [comment]
I can take this - not super familiar with the code base (yet) but I can at least open a PR taking a stab at it and can iterate from there.

**Comment by antoinetran** [comment]
Hi @tskillian , any news on if the interfaces for providers are created yet? :)
I do some dev for one provide (interlink), and have such a need to get the service account as projected volume.

**Comment by tskillian** [comment]
No, but I have [this PR](https://github.com/virtual-kubelet/virtual-kubelet/pull/1258) open to at least let providers take ownership of downward API resolution. That should let you do whatever it is you need to do.

**Comment by antoinetran** [comment]
> No, but I have [this PR](https://github.com/virtual-kubelet/virtual-kubelet/pull/1258) open to at least let providers take ownership of downward API resolution. That should let you do whatever it is you need to do.

@tskillian ok thanks for your answer! After reading the PR, I think up until now, virtual kubelet has parts of downward API implemented only for environment variables, and your PR is about skipping what virtual kubelet is actually doing. But nothing is done for downward API used in volume (like service account).

So my understanding is, there is no library nor helper (yet) related to downward API yet, but each provider can still do whatever in their own code.

**Comment by tskillian** [comment]
Ah I understand now - yea my understanding is also that the VK will not attempt any resolution of the downward API in volumes. My main concern is that if there's a downward API that isn't supported or can't be handled by VK's internal logic, the provider never gets called. For example, if a pod uses a node IP (not supported by VK), it'll just sit pending while scheduling to the VK indefinitely.
